### PR TITLE
[Fix] Show Alias Fields in Discover Tab 

### PR DESCRIPTION
**Pull request by sejli**
### Description

Adds a new advanced setting that updates the Discover's search request to include `"fields": ["*"]`. Defaults to off, open to suggestions for the title and description of the new setting.

Thanks @LDrago27 for the [deep dive](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7632#issuecomment-2307713526) and [fix](https://github.com/LDrago27/OpenSearch-Dashboards/commit/fddc1a933b6cbd5c2aa502a82d6de63603e3ef09)!

### Issues Resolved

#7632 

## Changelog

- fix: Show alias fields in Discover tab

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ashwin-pc — APPROVED** [review]

File: src/plugins/data/server/ui_settings.ts [hunk 1]
```diff
@@ -762,5 +762,19 @@ export function getUiSettings(): Record<string, UiSettingsParams<unknown>> {
...
+      }),
+      value: false,
+      description: i18n.translate('data.advancedSettings.searchUseFieldsDescription', {
+        defaultMessage: `
+        <strong>Experimental</strong>:
+        Adds the <code>"fields": ["*"]</code> property to search request body`,
+      }),
+      schema: schema.boolean(),
+      category: ['search'],
+      requiresPageReload: true,
```
Comment (joshuali925) [inline src/plugins/data/server/ui_settings.ts:777]: why does this require reload?

File: src/plugins/data/server/ui_settings.ts [hunk 2]
```diff
@@ -762,5 +762,19 @@ export function getUiSettings(): Record<string, UiSettingsParams<unknown>> {
       }),
       schema: schema.arrayOf(schema.string()),
     },
+    [UI_SETTINGS.SEARCH_USE_FIELDS]: {
+      name: i18n.translate('data.advancedSettings.searchUseFieldsTitle', {
+        defaultMessage: 'Include all fields in search request',
```
Comment (joshuali925) [inline src/plugins/data/server/ui_settings.ts:767]: this setting is to include all fields, would `search:includeAllFields` be better than `search:useFields`?

**Review by joshuali925 — DISMISSED** [review]

File: src/plugins/data/server/ui_settings.ts (hunk 1 above)
Comment (sejli) [reply src/plugins/data/server/ui_settings.ts:777]: Just checked with @kavilla, `search_source` should access the dependencies every time it calls `flatten()`, so we don't need it. Nice catch.

File: src/plugins/data/server/ui_settings.ts (hunk 2 above)
Comment (sejli) [reply src/plugins/data/server/ui_settings.ts:767]: My thinking was that previously, we weren't using the `fields` field in the request body. Turning on this property "uses" `fields`. I think in this case, the title itself could be something different.

Comment (ashwin-pc) [reply src/plugins/data/server/ui_settings.ts:767]: I like this suggestion

**Review by kavilla — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Pull request merged**